test/tests/performance/pbs_cgroups_stress.py

```diff
@@ -0,0 +1,224 @@
+                                      'lib',
+                                      'python',
+                                      'altair',
+                                      'pbs_hooks',
+                                      'pbs_cgroups.PY')
+        self.load_hook(self.hook_file)
+        # Enable the cgroups hook
+        conf = {'enabled': 'True', 'freq': 2}
+        self.server.manager(MGR_CMD_SET, HOOK, conf, self.hook_name)
+        # Restart mom so exechost_startup hook is run
+        self.mom.signal('-HUP')
+
+    @staticmethod
+    def get_paths():
+        """
+        Returns a dictionary containing the location where each cgroup
+        is mounted.
+        """
+        paths = {}
+        # Loop through the mounts and collect the ones for cgroups
+        with open(os.path.join(os.sep, "proc", "mounts"), 'r') as fd:
+            for line in fd:
+                entries = line.split()
+                if entries[2] != "cgroup":
+                    continue
+                flags = entries[3].split(',')
+                subsys = os.path.basename(entries[1])
+                paths[subsys] = entries[1]
+                if 'memory' in flags:
+                    paths['memsw'] = paths[subsys]
+                    paths['memory'] = paths[subsys]
+        return paths
+
+    def load_hook(self, filename):
+        """
+        Import and enable a hook pointed to by the URL specified.
+        """
+        try:
+            with open(filename, 'r') as fd:
+                script = fd.read()
+        except IOError:
+            self.assertTrue(False, "Failed to open hook file %s" % filename)
+        events = '"execjob_begin,execjob_launch,execjob_attach,'
+        events += 'execjob_epilogue,execjob_end,exechost_startup,'
+        events += 'exechost_periodic"'
+        a = {'enabled': 'True',
+             'freq': '2',
+             'event': events}
+        self.server.create_import_hook(self.hook_name, a, script,
+                                       overwrite=True)
+        # Add the configuration
+        self.load_config(self.cfg0)
+
+    def load_config(self, cfg):
+        """
+        Create a hook configuration file with the provided contents.
+        """
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, cfg)
+        os.close(fd)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': fn}
+        self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(fn)
+        self.mom.log_match('pbs_cgroups.CF;copy hook-related ' +
+                           'file request received',
+                           max_attempts=5,
+                           starttime=self.server.ctime)
+        self.logger.info("Current config: %s" % cfg)
+        # Restart MoM to work around PP-993
+        self.mom.restart()
+
+    @timeout(600)
+    def test_cgroups_epilogue_race(self):
```

Comment:
The race condition applies to other events as well, not just epilogue. So the test name is inaccurate. `test_cgroup_race_condition` would be better.

Reply:
Renamed